src/main/java/com/aws/greengrass/clientdevices/auth/configuration/RuntimeConfiguration.java

```diff
@@ -66,4 +72,31 @@ public void updateCAPassphrase(String passphrase) {
         caPassphrase.withValue(passphrase);
     }
 
-}
\ No newline at end of file
+    /**
+     * Update (or create if not available) opaque client device thing details in the config.
+     *
+     * @param thingName    client device Thing name
+     * @param thingDetails opaque client device Thing details
+     */
+    public void updateClientDeviceThing(String thingName, Map<String, Object> thingDetails) {
+        Topics clientDeviceThings = config.lookupTopics(CLIENT_DEVICE_THINGS_KEY);
+        Topics clientDeviceThing = clientDeviceThings.lookupTopics(thingName);
+        clientDeviceThing.replaceAndWait(thingDetails);
+    }
+
+    /**
+     * Retrieves opaque client device thing details from the config.
+     *
+     * @param thingName Thing name
+     * @return map of thing details
+     */
+    public Map<String, Object> getClientDeviceThing(String thingName) {
+        Topics clientDeviceThings = config.findTopics(CLIENT_DEVICE_THINGS_KEY);
+        if (clientDeviceThings == null || clientDeviceThings.findTopics(thingName) == null
+                || clientDeviceThings.findTopics(thingName).isEmpty()) {
+            return Collections.emptyMap();
+        } else {
+            return clientDeviceThings.findTopics(thingName).toPOJO();
```

Comment:
Do we need to do this? This will be copying data into a map. Would be best if we don't need to duplicate data.

Reply:
right, refactored to return `Topics` directly, instead of a map.